engine/worker/builtin_junit.go

```diff
@@ -85,28 +84,70 @@ func runParseJunitTestResultAction(a *sdk.Action, pbJob sdk.PipelineBuildJob, st
 	// update global stats
 	for _, ts := range v.TestSuites {
```

Comment:
may this code be in a single function; and have unit tests (table unit tests) on this algorithm